ledger/src/sigverify_shreds.rs

```diff
@@ -137,7 +137,7 @@ fn slot_key_data_for_gpu<
     let keyvec_size = keys_to_slots.len() * size_of::<T>();
     keyvec.resize(keyvec_size, 0);
 
-    for (i, (k, slots)) in keys_to_slots.iter_mut().enumerate() {
```

Comment:
Fixes this:

```
error: mutable key type
   --> ledger/src/sigverify_shreds.rs:135:5
    |
135 |     let mut slot_to_key_ix = HashMap::new();
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[deny(clippy::mutable_key_type)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#mutable_key_type
```

[`slot_to_key_ix`](https://github.com/solana-labs/solana/pull/8199/files#diff-a05adb75747668527389bb08feb77227L135)'s type is inferred as `HashMap<&mut Slot, usize>` because of this `iter_mut()`.